### ISSUE TITLE
[UX] Add newly-created roles between the "Authenticated" and "Administrator" roles

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#4507]
This was suggested by @BWPanda in https://github.com/backdrop/backdrop-issues/issues/4504#issuecomment-665393999

> ...creating a new role adds it to the end of the list, after 'Administrator'. Assuming Drupal's recommendations about ordering apply to Backdrop too, maybe new roles should be placed between Authenticated and Administrator...?

The recommendation @BWPanda mentions above is a portion of the help text that was added as part of #4504:

> To make the permissions page easier to navigate, we recommended ordering roles from the least permissive (Anonymous) to the most permissive (usually Administrator).

**Comment by klonos** [comment]
I initially attempted to implement this in a very simplistic way, by adding all new roles after the Authenticated role (setting their weight to `'weight' => $form_state['values']['roles'][BACKDROP_AUTHENTICATED_ROLE]['weight'] + 1`), but testing by @jenlampton proved that this does not work as expected:

>This doesn't behave reliably the way it's currently implemented. The first time you add new roles it's fine, because all the weights for the new roles are the same, so it places them alphabetically. But if you then re-order the roles, then try to add another new role, it behaves very strangely.
>
>I think rather than trying to put it one position after the `Authenticated` role, we should try to place it one position before the `Administrator` role. Not only does this more closely match what people will be doing initially (since people would usually add more than one role in increasing order of permissions) but it will also work nicely when people return at a later date and add another new role. 
>
>From an implementation standpoint, we'll need to pull all the current roles by weight, insert the new one at the position where we'd like it, then save all the roles that have new weights. If we are assuming most people leave the `Administrator` role as the most permissive, in most cases this would mean only saving the new role, and the `Administrator` role. Alternatively, to place a role after the `Authenticated` role, we'd likely need to save them all. 
>
>Since this is no longer a simple change, I recommend we do this in another issue and keep this one to the text change. :)

**Comment by albanycomputers** [comment]
Thanks @BWPanda I came to that conclusion :)

I would suggest that the Anon, auth and admin users be "Locked" into position so that they can't be repositioned with drag-handles :)

**Comment by BWPanda** [comment]
I agree re. Anon and Auth, but not Admin. There's nothing special about that role and no reason it can't be less permissive than a new, custom, more advanced role...

**Comment by bugfolder** [comment]
Also, in upgrades from D7, there's likely not a role with that title/machine name for Admin.

**Comment by klonos** [comment]
> I agree re. Anon and Auth, but not Admin.

Wondering if there is an alternative/intuitive way to replace the entire "Administrator role" fieldset with some sort of "drag the admin role to the bottom of the list" functionality 🤔  ...I'll play around with some mockups to see if that works well.
